extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeMixPlaylistExtractor.java

```diff
@@ -0,0 +1,175 @@
+    }
+
+    @Override
+    public void onFetchPage(@Nonnull Downloader downloader)
+        throws IOException, ExtractionException {
+        final String url = getUrl() + "&pbj=1";
+        final JsonArray ajaxJson = getJsonResponse(url, getExtractorLocalization());
+        JsonObject initialData = ajaxJson.getObject(3).getObject(RESPONSE);
+        playlistData = initialData.getObject(CONTENTS).getObject(TWO_COLUMN_WATCH_NEXT_RESULTS)
+            .getObject(PLAYLIST).getObject(PLAYLIST);
+    }
+
+    @Nonnull
+    @Override
+    public String getName() throws ParsingException {
+        final String name = playlistData.getString("title");
+        if (name == null) {
+            throw new ParsingException("Could not get playlist name");
+        }
+        return name;
+    }
+
+    @Override
+    public String getThumbnailUrl() throws ParsingException {
+        try {
+            final String playlistId = playlistData.getString("playlistId");
+            return getThumbnailUrlFromId(playlistId);
+        } catch (Exception e) {
+            throw new ParsingException("Could not get playlist thumbnail", e);
+        }
+    }
+
+    @Override
+    public String getBannerUrl() {
+        return "";
+    }
+
+    @Override
+    public String getUploaderUrl() {
+        //Youtube mix are auto-generated
+        return "";
+    }
+
+    @Override
+    public String getUploaderName() {
+        //Youtube mix are auto-generated by YouTube
+        return "YouTube";
+    }
+
+    @Override
+    public String getUploaderAvatarUrl() {
+        //Youtube mix are auto-generated by YouTube
+        return "https://upload.wikimedia.org/wikipedia/commons/thumb/0/09/YouTube_full-color_icon_(2017).svg/320px-YouTube_full-color_icon_(2017).svg.png";
+    }
+
+    @Override
+    public long getStreamCount() {
+        // Auto-generated playlist always start with 25 videos and are endless
+        return ListExtractor.ITEM_COUNT_INFINITE;
+    }
+
+    @Nonnull
+    @Override
+    public InfoItemsPage<StreamInfoItem> getInitialPage() throws ExtractionException {
+        StreamInfoItemsCollector collector = new StreamInfoItemsCollector(getServiceId());
+        collectStreamsFrom(collector, playlistData.getArray(CONTENTS));
+        return new InfoItemsPage<>(collector, getNextPageUrl());
+    }
+
+    @Override
+    public String getNextPageUrl() throws ExtractionException {
+        final JsonObject lastStream = ((JsonObject) playlistData.getArray(CONTENTS)
+            .get(playlistData.getArray(CONTENTS).size() - 1));
+        if (lastStream == null || lastStream.getObject(PLAYLIST_PANEL_VIDEO_RENDERER) == null) {
+            throw new ExtractionException("Could not extract next page url");
+        }
+        return "https://youtube.com" + lastStream.getObject(PLAYLIST_PANEL_VIDEO_RENDERER)
+            .getObject("navigationEndpoint").getObject("commandMetadata")
+            .getObject("webCommandMetadata").getString("url") + "&pbj=1";
```

Comment:
You should call `getUrlFromNavigationEndpoint()` and just append `&pbj=1`. You'll first need to modify `getUrlFromNavigationEndpoint()` so that it handles the `index` parameter.

Reply:
I just saw that I already handled the index incorrectly, since i assumed the index in the url would count upwards. But it always stays at 25.

Reply:
Is there a reason for using `&amq;` instead of just `&` since this is throwing a malformed url exception for me when i create new extractor with the return value of `getUrlFromNavigationEndpoint()`

https://github.com/TeamNewPipe/NewPipeExtractor/blob/dev/extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler/YoutubeParsingHelper.java#L302

Reply:
Yes, there's a reason, but it's wrong. That function should return it with `&`, which should then be replaced with `&amp;` at https://github.com/TeamNewPipe/NewPipeExtractor/blob/dev/extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler/YoutubeParsingHelper.java#L319.